### PR TITLE
add redux for basket

### PR DESCRIPTION
**Pull request by gomelviiv**

### CODE REVIEW

File: src/redux/reducers/basket/actionTypes.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const ADD_JEWELRY_TO_BASKET = 'ADD_JEWELRY_TO_BASKET';
```
Comment (angelinabylitskaya) [inline src/redux/reducers/basket/actionTypes.tsx:1]: Here an example of how to create typed actions:
```
const typesNames = [
  'LOGIN',
  'LOGIN_SUCCESS',
  'LOGIN_FAILURE',
  'LOGOUT',
  'CHECK_AUTH',
  'SET_AUTH',
  'GET_CURRENT_USER_SUCCESS',
  'GET_CURRENT_USER_FAILED',
  'DELETE_ERROR_MESSAGE',
] as const;

export type AuthActionType = {
  [Key in typeof typesNames[number]]: string;
};

export const prefix: string = 'AUTH';

export const authActionTypes: AuthActionType = createConstants<AuthActionType>(
  typesNames,
  prefix,
);

```
createConstants:
```
export const createConstants = <T>(
  constants: readonly any[],
  constantType: string,
): T =>
  constants.reduce((acc, constant: string) => {
    const prefix = constantType ? `${constantType}/` : '';
    acc[constant] = `${prefix}${constant}`;
    return acc;
  }, {});
```

All actions will be available inside authActionTypes object.

File: src/redux/reducers/basket/index.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    (sum: number, key: any) => newItems[key].items.length + sum,
+    0,
+  );
+};
+
+const getTotalPrice = (newItems: any) => {
+  return Object.keys(newItems).reduce((sum: number, key: any) => newItems[key].totalPrice + sum, 0);
+};
+
+const basketReducer = (state: IBasket = initialState, action: IBasketPayloadType<any>): IBasket => {
```
Comment (angelinabylitskaya) [inline src/redux/reducers/basket/index.tsx:11]: Move reducer to a separate file.

File: src/redux/reducers/basket/index.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { IBasket, IBasketItem, IBasketPayloadType } from './type';
+import { IJewelryItem } from '../jewelry/type';
+
+const initialState = {
+  totalCount: 0,
+  totalPrice: 0,
+  items: {},
+} as IBasket;
+
+const getTotalPriceNewItem = (arr: Array<IJewelryItem>) =>
```
Comment (angelinabylitskaya) [inline src/redux/reducers/basket/index.tsx]: Move helpers to a separate file.

File: src/redux/reducers/basket/type.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { IJewelryItem } from '../jewelry/type';
```
Comment (angelinabylitskaya) [inline src/redux/reducers/basket/type.tsx:1]: Keep creating a file per entity.
Better to keep the following structure:
redux(or store or root-store)/
- features (a folder with all store features)/
--- basket (feature)/
----- index.ts (module export file)
----- reducers.ts
----- types.ts
----- actions.ts
----- selectors.ts
----- state.ts (initial state)
----- sagas.ts (if used)

**Review by angelinabylitskaya — CHANGES_REQUESTED** [review]

**Comment by gomelviiv** [comment]
@angelinabylitskaya, i tried to fix all issues.

**Pull request merged**